src/dialogs/confirmation-dialog.ts

```diff
@@ -8,9 +8,14 @@ import { classMap } from "lit/directives/class-map.js";
 @customElement("esphome-confirmation-dialog")
 class ESPHomeConfirmationDialog extends LitElement {
   @state() private _params?: ConfirmationDialogParams;
+  @state() private _resolve?: (value: boolean | PromiseLike<boolean>) => void;
```

Comment:
You can change these to be. They are always set if dialog is open. Only keep `if (!this._params) {` inside `render`
```suggestion
  @state() private _params!: ConfirmationDialogParams;
  @state() private _resolve!: (value: boolean | PromiseLike<boolean>) => void;
```

Reply:
done